### ISSUE TITLE
fixdeps.js symlink error Win7 npm 3

### ISSUE THREAD

**Issue by gwicksted** [theintern/intern#500]
Possibly related to #429.

Node `4.1.0` with npm `3.3.3` on `Windows 7 x64`

```sh
npm ERR! Windows_NT 6.1.7601
npm ERR! argv "C:\\Program Files (x86)\\nodejs\\node.exe" "C:\\Users\\myusername\\AppData\\Roaming\\npm\\node_modules\\
npm\\bin\\npm-cli.js" "install" "intern" "--save-dev"
npm ERR! node v4.1.0
npm ERR! npm  v3.3.3
npm ERR! code ELIFECYCLE

npm ERR! intern@3.0.3 install: `node support/fixdeps.js`
npm ERR! Exit status 1
```

Further up, more details revealed:

```sh
fs.js:902
  return binding.symlink(preprocessSymlinkDestination(destination, type, path),
                 ^

Error: EPERM: operation not permitted, symlink 'D:\dev\myprojectfolder\node_modules\dojo' -> 'D:\dev\myprojectfolder\node_modules\intern\node_modules\dojo'
    at Error (native)
    at Object.fs.symlinkSync (fs.js:902:18)
    at D:\dev\myprojectfolder\node_modules\intern\support\fixdeps.js:33:6
    at Array.forEach (native)
    at Object.<anonymous> (D:\dev\myprojectfolder\node_modules\intern\support\fixdeps.js:14:28)
    at Module._compile (module.js:434:26)
    at Object.Module._extensions..js (module.js:452:10)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
    at Function.Module.runMain (module.js:475:10)
```

**Comment by gwicksted** [comment]
Paths need normalizing before performing a string comparison for equality.  The two offenders:

```js
var expected = path.join(__dirname, '..', 'node_modules');
```

```js
	var actualPath = path.dirname(require.resolve(path.join(dependency, 'package.json')));
```

a `path.normalize` ensures that "C:\" and "c:\" will be the same case before performing string comparisons.  I believe it's __dirname that returns the inconsistent case.

**Comment by csnover** [comment]
Copying over from the closed PR #501…

The issue here is:

1. npm 3 doesn’t ever install dependencies into the intern/node_modules` directory so everyone always ends up doing symlinks
2. By default, non-administrative users in Windows don’t have permissions to symlink directories

So, you’ll need to run as admin when installing with npm 3 on Windows. 

There are two possible fixes:

1. Change 'dir' to 'junction' in the symlink call, since it seems creating a junction doesn’t require extra permissions for some reason. Junctions have to be absolute directories, but these symlinks are already absolute so this should not matter.
2. Catch the EPERM error and copy instead of symlinking.
